### PR TITLE
PEN-1496: [CMG] Ability to configure components in navigation bar

### PR DESCRIPTION
**Pull request by vinnie-olsen**
**If you have not filled out the checklist below, the pr is not ready for review.**

## Description
Added configuration custom fields (with both "desktop" and "mobile" options) to the Header Nav Chain Block.

## Jira Ticket
- [PEN-1496](https://arcpublishing.atlassian.net/browse/PEN-1496)

## Acceptance Criteria
image

## Test Steps
1. In `Fusion-News-Theme` repo, checkout `add-fake-weather-feature-for-nav-testing` & `git pull` (https://github.com/WPMedia/Fusion-News-Theme/pull/152/files)
2. After checking out this feature branch in `fusion-news-theme-blocks`, run `rm -rf ./node_modules && npx lerna clean && npm i && npx lerna bootstrap && cd blocks/header-nav-chain-block && npm i && cd ../..`
3. Run `npx fusion start -f -l @wpmedia/header-nav-chain-block,@wpmedia/shared-styles` in `Fusion-News-Theme`
4. Open homepage in PB editor and test three different combinations of settings. After each change, stage & publish page and open http://localhost/pf/homepage/?_website=the-dagen in separate tab:
a. Logo left-aligned **without** horizontal links defined
b. Logo left-aligned **with** horizontal links defined
c. Logo center-aligned (with or without horizontal links defined - shouldn't matter but feel free to test both)
d. Now within the mobile and desktop groups (under custom fields), try various different combinations of buttons placed in the left and right nav bar containers. When testing the "custom" type, you will need to add a feature as a child of the nav bar chain. Then, you will need to specify the position of the child component to use (which is a "1-based" index - not zero-based). 
image
5. Test clicking on search button and search input box should expand (pushing other elements to the right). The search box should have the same height as the other nav bar buttons.
6. Test making the viewport smaller which cuts off space for the horizontal links to display. When a horizontal links bar has too many links to fit on the current viewport width, the horizontal link elements should be truncated as needed (disappear when no space available for them). Horizontal links bar should be left-aligned against the logo.
7. Ensure that everything displays as expected across breakpoints. When the logo is centered, it should display pretty much the same on tablet/mobile (no horizontal links) except mobile only allows one button/widget on each side of the nav. When the logo is left-aligned, the horizontal links should be hidden on the mobile breakpoint.

### Before
There were no configuration options for how buttons/widgets could be placed (right vs left side, item order).

### After
There are now configuration options for how buttons/widgets can be placed (right vs left side, item order). The new options allow you to configure what items should display (and in what order) based on the breakpoint (desktop vs mobile).

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [X] Confirmed all the test steps a reviewer will follow above are working. 
- [X] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [X] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [X] Confirmed this PR has unit test files
  - [X] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [X] Confirmed relevant documentation has been updated/added.

### CODE REVIEW

File: blocks/header-nav-chain-block/chains/header-nav-chain-block/default.jsx [hunk 1]
```diff
@@ -170,6 +154,20 @@ const Nav = (props) => {
...
+    };
+
+    window.addEventListener('keydown', handleEscKey, true);
+    return () => {
+      window.removeEventListener('keydown', handleEscKey);
+    };
+  }, []);
+
   useEffect(() => {
     const mastHead = document.querySelector('.masthead-block-container .masthead-block-logo');
```
Comment (brentmiller1973) [inline blocks/header-nav-chain-block/chains/header-nav-chain-block/default.jsx:172]: Wondering if this should be accessed via a ref at this point?  Don't think however its urgent right now

**Review by brentmiller1973 — APPROVED** [review]
Nice man and great test coverage

**Comment by vinnie-olsen** [comment]
Thanks Brent!

File: blocks/header-nav-chain-block/chains/header-nav-chain-block/default.jsx (hunk 1 above)
Comment (vinnie-olsen) [reply blocks/header-nav-chain-block/chains/header-nav-chain-block/default.jsx:172]: Yeah I completely agree 🙂 We can definitely address that in a follow-up ticket at some point. I think there's some additional refactoring and clean-up that we could benefit from once we have some time.

**Pull request merged**